graph/pull.go

```diff
@@ -112,7 +112,7 @@ func (s *TagStore) Pull(image string, tag string, imagePullConfig *ImagePullConf
 			lastErr = err
 			continue
 		}
-		if fallback, err := puller.Pull(tag); err != nil {
+		if fallback, err := puller.Pull(tag, dryRun); err != nil {
```

Comment:
This doesn't make a lot of sense to have an interface with a `dryRun` parameter. Just change the implementation of the interface to make it not take an action.

The main problem with this is that one must always consider how `dryRun` is interpreted when making additions to the implementation. If an individual is fixing a bug in an implementation and is not aware of this interface, we risk having unprotected mutations.

The right approach is to have a separate implementation of `puller` resolved at instantiation time that implements puller. If we have to parameterize on deeper methods of puller, we should provide interfaces that can be injected further down and do not provide a no-op.